src/foam/core/controller/ApplicationController.js

```diff
@@ -858,7 +858,7 @@ foam.CLASS({
       if ( hash ) hash = hash.substring(1);
       if ( ! hash || hash == 'null' /* How does it even get set to null? */ ) {
         await this.pushDefaultMenu();
-      } else if ( hash != this.currentMenu?.id || this.currentMenu.authenticate ) {
+      } else if ( hash != this.currentMenu?.id || this.currentMenu.authenticate || hash == 'sign-in' ) {
```

Comment:
Dont think this is acceptable, we would need to do this for all unauthenticated menu

Reply:
why, we are not going to call `onUserAgentAndGroupLoaded()` whenever we push any unauthenticated menu, do we?

Reply:
Why was the original change made? https://github.com/kgrgreer/foam3/pull/3742
This is intended behaviour, Unauthed menus are not typically used by a logged in user

Reply:
I see it is for error pages, we might need a more nuanced flag than authenticate then, maybe a three way enum, unauth, auth and both menus

Reply:
`onUserAgentAndGroupLoaded()` is only called after successfully signin via ClientLoginService and in ApplicationController.onClientLoad() for when the subject real user is not anonymous, I don't see it is being called elsewhere or did I miss something?

Reply:
Also if you signin through a wizard would we want to keep the user in the wizard flow or immediately send the user to the default menu?

Reply:
> Why was the original change made? #3742 This is intended behaviour, Unauthed menus are not typically used by a logged in user

I think unauthed menus should generally be useable by logged in users. For the typical web-site-like site, this would be the norm where some parts are protected and the rest is public. If we didn't want logged in users to see some menus we could give anonymousUser the permission but not regular users.

Reply:
#3742 allows unauthed menus to be useable by logged in users if they have have permission. 
Before the users (although having permission) will still be redirected to default menu when trying to access an unauthed menu.